mantidproject/mantid
# Indirect - Move the run button in Data Corrections and disable when running

**Description of work.**
This PR moves the `Run` button in the Indirect `Data Corrections` interface so that it is above the `Output` options. This allows the `Run` button to be disabled while a process is running.
The `Plot` button is also now disabled while plotting is being done.

**To test:**
1. `Interfaces`->`Indirect`->`Data Corrections`-> All Tabs
2. Load and enter the data below for the respective tabs.
3. You should try observe several things:
- Click Run: The `Run`, `Plot` and `Save` buttons should all be disabled while running. The text on the `Run` button should also change to `Running...`
                     
- Click Plot: The `Run`, `Plot` and `Save` buttons should all be disabled while plotting (If the plotting is quick this may be hard to notice). The text on the `Plot` button should also change to `Plotting...`
                    

#### **Data** ####
**Note**: the data provided may not be the best, but we just need to check how the `Run` and `Plot` buttons  behave

**Container Subtraction:**
Sample: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2460924/irs26176_graphite002_red.zip)
Container: [irs26174_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2460923/irs26174_graphite002_red.zip)

**Calculate Paalman Pings:**
Sample: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2460924/irs26176_graphite002_red.zip)
Sample Thickness: `0.1cm`
Chemical Formula: `H2-O`

**Calculate Monte Carlo Absorption:**
Sample: `26176` (on the system)
Container: `26174` (on the system)
Shape: `FlatPlate`
Sample and Container Height & width: `1.0`
Sample and Container Thickness: `0.1`
Sample mass density: `1`
Sample Formula: `H2-O`
Container mass density: `6`
Container Formula: `V`
It may be a long wait when running! The output workspace named 

**Apply Absorption Corrections:**
Sample: [irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2460924/irs26176_graphite002_red.zip)
Corrections: [irs26176_graphite002_FlatPlate_MC_Corrections.zip](https://github.com/mantidproject/mantid/files/2464168/irs26176_graphite002_FlatPlate_MC_Corrections.zip)
Tick `Use Container`
Container: [irs26174_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2460923/irs26174_graphite002_red.zip)

Fixes #23700 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
